content/2019/Making-things-flow/index.md

```diff
@@ -0,0 +1,98 @@
+
+Here's a [gist][instant-grep] that showcases this. This is also behaviour is also available in [vim-qf][vim-qf].
+
+This method is very useful as there are many ways the quickfix list may be populated and yet we are able to create a behaviour that is consistent with a self contained implementation.
+
+## Don't disrupt my flow
+
+Disabling `hlsearch` is a common operation for users that have `set hlsearch` as after performing the search and finding the desired result the highlighting becomes superfluous. To ease this many Vim users have a mapping to deal with this. But that's an extra step that the user has to think about. Instead we can alter the behaviour of Vim to intelligently disable and re-enable `hlsearch` as done with [vim-cool][vim-cool].
+
+As noted in the README this plugin really became "cool" with [this][vim-cool_purpleP]. It doesn't change the intention of the plugin but it achieves it in a far robust and elegant way.
+
+## Don't break my flow
+
+Another example is the movement of the cursor when dealing with operators. Vim has moves the cursor after invoking operators and I'd prefer it didn't always do so. Here we wish to prevent a behaviour default to Vim rather than craft a complementary one.
+
+Avoiding this in the case of operator mappings can be particularly ugly. A common method is to drop a mark or save a view in the mapping. However the mapping must end with `g@` so Vim will wait for the operator, meaning the mapping itself can't invoke the cursor movement back to its original location. As such the movement would have to be placed inside the function that `operatorfunc` has been set to. Note that dropping a mark or saving a view inside the `operatorfunc` function is not an option as the cursor has already been moved by the time this is invoked.
+
+Functions should not require such a priori knowledge about the mappings that are calling them.
+
+A solution to this is to save a view whenever `operatorfunc` is set. Then whenever the cursor is moved check if `operatorfunc` is set, if so restore the view, dispose of the temporary variable containing said view, and unset `operatorfunc` with autocmd's disabled as to avoid a loop.
+
+```vim
+function! OpfuncSteady()
+  if !empty(&operatorfunc)
+    call winrestview(w:opfuncview)
+    unlet w:opfuncview
+    noautocmd set operatorfunc=
+  endif
+endfunction
+
+augroup OpfuncSteady
+  autocmd!
+  autocmd OptionSet operatorfunc let w:opfuncview = winsaveview()
+  autocmd CursorMoved * call OpfuncSteady()
+augroup END
+```
+
+Again it's the transparency that makes this neat. With this snippet neither the user or even the rest of their configuration have to think about this. The behaviour of the editor has simply been altered.
+
+## Conclusion
+
+I hope this has showcased some interesting ideas about scripting Vim's behaviour. I've placed and emphasis on "flow" as the examples shown aim to have little to zero cognitive impact.
+
+---
+
+_License notice_
```

Comment:
Could you add the actual license?

Reply:
BTW, I just noticed it's also missing in your article on colors.

Reply:
I think more than just mine are missing this. Is there a preferred license?

Reply:
It's your "content" so you are in a better position than me to decide. [This one](https://creativecommons.org/licenses/by-nc-sa/3.0/) seems interesting.